doc/introduction/interfaces.rst

```diff
@@ -3,17 +3,104 @@
 
 .. _intro_interfaces:
 
-Interfaces and training
-=======================
+Gradients and training
+======================
 
 PennyLane offers seamless integration between classical and quantum computations. Code up quantum
-circuits in PennyLane, and connect them easily to the top scientific computing and machine learning
-libraries.
+circuits in PennyLane, compute :doc:`gradients of quantum circuits <glossary/quantum_gradient>`, and
+connect them easily to the top scientific computing and machine learning libraries.
+
+Gradients
+---------
+
+When creating a QNode, you can specify the :doc:`differentiation method
+<glossary/quantum_differentiable_programming>` that PennyLane should use whenever the gradient of
+that QNode is requested.
+
+.. code-block:: python
+
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.probs(wires=0)
+
+PennyLane currently provides the following differentiation methods for QNodes:
+
+Simulation-based differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods use `reverse accumulation
+<https://en.wikipedia.org/wiki/Automatic_differentiation#Reverse_accumulation>`__ to compute
+gradients; a well-known example of this approach is backpropagation. These methods are **not** hardware compatible; they are only supported on
+*statevector* simulator devices such as :class:`default.qubit <~.DefaultQubit>`.
+
+However, for rapid prototyping on simulators, these methods typically out-perform forward-mode
+accumulators such as the parameter-shift rule and finite-differences. For more details, see the
+:doc:`quantum backpropagation <demos/tutorial_backprop>` demonstration.
+
+* ``"backprop"``: Use standard backpropagation.
+
+  This differentiation method is only allowed on simulator
+  devices that are classically end-to-end differentiable, for example
+  :class:`default.qubit <~.DefaultQubit>`. This method does *not* work on devices
+  that estimate measurement statistics using a finite number of shots; please use
+  the ``parameter-shift`` rule instead.
+
+* ``"adjoint"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  The `adjoint method <https://arxiv.org/abs/2009.02823>`__  reverses through the circuit after a
+  forward pass by iteratively applying the inverse (adjoint) gate. This method is similar to
+  ``"backprop"``, but trades off increased runtime with significantly lower memory usage.
+
+* ``"reversible"``: A form of backpropagation that takes advantage of the unitary or reversible
```

Comment:
```suggestion
* ``"reversible"``: A form of differentiation that takes advantage of the unitary or reversible
```
Is it a form of backprop? (It might be, maybe I'm just confused)

Reply:
Ah... I believe that `reversible` is a form of backpropagation, in that backpropagation refers to reverse-mode gradient accumulation of a scalar-valued function?